src/framework/components/anim/system.js

```diff
@@ -44,6 +43,9 @@ class AnimComponentSystem extends ComponentSystem {
         if (data.animationAssets) {
             component.animationAssets = Object.assign(component.data.animationAssets, data.animationAssets);
         }
+        if (data.rootBone) {
+            component.rootBone = data.rootBone;
+        }
     }
 
     onAnimationUpdate(dt) {
```

Comment:
Not public without JSDoc, same for `initializeComponentData` method right above

Reply:
These methods aren't public on any other components, so i'll keep them private for now.

Reply:
Is PC Viewer bad style then? It is used publicly here: https://github.com/playcanvas/playcanvas-viewer/blob/master/src/viewer.ts#L802

All these private/non-documented methods cause dozens of type errors across projects.

Reply:
Yeah I should really have added a comment at that line in the viewer. It's not a pattern we publicly support but it was useful for the viewers rather particular timeline scrubber.